Dockerfile

```diff
@@ -66,21 +66,26 @@ ARG TARGETARCH
 RUN set -ex; \
     if [ "$PG_MAJOR_VERSION" -ge 16 ] && [ "$TARGETARCH" != "arm" ]; then \
         apk update; \
+        # install shared libraries required by pyarrow needed at runtime
+        apk add libarrow libparquet; \
+        # install required dependencies for building pyarrow from source
         apk add --no-cache --virtual .pgai-deps \
             git \
             build-base \
             cargo \
             cmake \
             python3-dev \
-            apache-arrow-dev \
-            py3-pip; \
+            py3-pip \
+            apache-arrow-dev; \
+        # pgai requires pip 23.0.1 or greater due to the use of --break-system-packages flag
+        if [ "$(pip --version | awk '{print $2; exit}')" \< "23.0.1" ]; then \
+            python3 -m pip install --upgrade pip==23.0.1; \
+        fi; \
         git clone --branch ${PGAI_VERSION} https://github.com/timescale/pgai.git /build/pgai; \
         cd /build/pgai; \
-        # note: this is a hack. pyarrow will be built from source, so must be pinned to this arrow version \
-        echo pyarrow==$(pkg-config --modversion arrow) >> ./projects/extension/requirements.txt; \
```

Comment:
This is not needed anymore since we get the headers from the `apache-arrow-dev` package and install from source using them. Am I right or perhaps missing something @JamesGuthrie ?

Reply:
This is still needed. Adding this to the requirements ensures that we install the version of pyarrow which corresponds to the version of libarrow which is available on the current platform.

Reply:
> This is still needed. Adding this to the requirements ensures that we install the version of pyarrow which corresponds to the version of libarrow which is available on the current platform.

Good point 💯